File: backend/app/models/manager_activity.py
# Copyright (c) 2026 Laurent Barbe
# Licensed under the Apache License, Version 2.0
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ManagerActivityEntry(BaseModel):
    id: int
    created_at: datetime
    action: str
    entity_type: Optional[str] = None
    entity_id: Optional[str] = None
    account_id: Optional[int] = None
    account_name: Optional[str] = None
    status: str
    user_email: str

File: frontend/src/api/managerActivity.ts
/*
 * Copyright (c) 2026 Laurent Barbe
 * Licensed under the Apache License, Version 2.0
 */
import { S3AccountSelector, withS3AccountParam } from "./accountParams";
import client from "./client";

export type ManagerActivityEntry = {
  id: number;
  created_at: string;
  action: string;
  entity_type?: string | null;
  entity_id?: string | null;
  account_id?: number | null;
  account_name?: string | null;
  status: string;
  user_email: string;
};

export async function listManagerActivity(
  accountId: S3AccountSelector,
  options?: { limit?: number }
): Promise<ManagerActivityEntry[]> {
  const { data } = await client.get<ManagerActivityEntry[]>("/manager/activity", {
    params: withS3AccountParam({ limit: options?.limit ?? 5 }, accountId),
  });
  return data;
}
